### ISSUE TITLE
Branch Protections should be synced on branch creation events

### ISSUE THREAD

**Issue by L1ghtman2k** [github/safe-settings#491]
Please describe the desired new functionality:

safe-settings should be able to handle [create](https://docs.github.com/en/webhooks-and-events/events/github-event-types#createevent)/[delete](https://docs.github.com/en/webhooks-and-events/events/github-event-types#deleteevent) events, and update branch protection rules(sync repo) for those events.

Ideally, there should be some sort of filter to ignore spammy branches, like ignoring dependabot_*.

Example usecase:
We would like to store repository metadata in .organization_name branch and make sure that branch is protected. However, this doesn't get triggered unless there is a cron, or some other sync that is scheduled

**Comment by decyjphr** [comment]
Basically listen to these events and act appropriately, right?
<img width="660" alt="image" src="https://github.com/github/safe-settings/assets/57544838/cd7214fc-e7d5-4110-8ac4-814374d5ddab">

**Comment by L1ghtman2k** [comment]
Correct, though, for very large orgs it would be nice to have a way to ignore certain branches (aka, dependabot)
